include/fiction/technology/charge_distribution_surface.hpp

```diff
@@ -0,0 +1,112 @@
+//
+// Created by Jan Drewniok on 23.11.22.
+//
+
+#ifndef FICTION_CHARGE_DISTRIBUTION_SURFACE_HPP
+#define FICTION_CHARGE_DISTRIBUTION_SURFACE_HPP
+
+#include "fiction/technology/cell_technologies.hpp"
+#include "fiction/technology/sidb_defects.hpp"
+#include "fiction/traits.hpp"
+
+#include <unordered_map>
+
+namespace fiction
+{
+
+/**
+ * Possible SiDB charges.
+ */
+enum class sidb_charge_states
+{
+    NONE,  // assigned when layout cell is empty
+    POSITIVE,
+    NEUTRAL,
+    NEGATIVE
+};
+
+struct sidb_charge
+{
+    /**
+     * Standard constructor.
+     */
+    constexpr explicit sidb_charge(const sidb_charge_states defect_type = sidb_charge_states::NONE) noexcept :
+            charge_state{defect_type}
+    {}
+
+    const sidb_charge_states charge_state;
+};
+
```

Comment:
Please add nested overloads as in `sidb_surface` to optimize the use of `charge_distribution_surface<charge_distribution_surface>`. You can create according traits in `traits.hpp`